public/app/plugins/datasource/loki/configuration/DerivedField.tsx

```diff
@@ -70,16 +76,50 @@ export const DerivedField = (props: Props) => {
   return (
     <div className={className} data-testid="derived-field">
       <div className="gf-form">
+        {config.featureToggles.lokiDerivedFieldsFromLabels && (
+          <Field
+            className={styles.nameMatcherField}
+            label={
+              <TooltipLabel
+                label="Type"
+                content="Derived fields can be based on a label or a regular expression base on the log message."
+              />
+            }
+          >
+            <Select
+              options={[
+                { label: 'Regex in log line', value: 'regex' },
+                { label: 'Label', value: 'label' },
+              ]}
+              value={fieldType}
+              onChange={(type) => {
+                if (type.value === 'label' || type.value === 'regex') {
+                  setFieldType(type.value);
+                  onChange({
+                    ...value,
+                    labelMatcher: type.value === 'label',
```

Comment:
I am wondering if we don't want to create a `matcherType = 'label' | 'regex'` in here. So it is more clear and in a case we want to expand to other option, it would be easy to do. 🤔

Reply:
Good point, done in 24062237a952e301636f37b45a8c4e021bdf8905